apps/aecore/src/aec_spend_tx.erl

```diff
@@ -114,16 +118,20 @@ signers(#spend_tx{sender = SenderPubKey}) -> [SenderPubKey].
 
 -spec process(tx(), aetx:tx_context(), aec_trees:trees(), height(), non_neg_integer()) -> {ok, aec_trees:trees()}.
 process(#spend_tx{sender = SenderPubkey,
-                  recipient = RecipientPubkey,
+                  recipient = RecipientPubkeyOrName,
                   amount = Amount,
                   fee = Fee,
                   nonce = Nonce}, _Context, Trees0, Height, _ConsensusVersion) ->
     AccountsTrees0 = aec_trees:accounts(Trees0),
+    NSTrees0 = aec_trees:ns(Trees0),
 
     {value, SenderAccount0} = aec_accounts_trees:lookup(SenderPubkey, AccountsTrees0),
     {ok, SenderAccount} = aec_accounts:spend(SenderAccount0, Amount + Fee, Nonce, Height),
     AccountsTrees1 = aec_accounts_trees:enter(SenderAccount, AccountsTrees0),
 
+    % possible errors have been filtered previously which check -> ensure_account_at_height
```

Comment:
with check?

Reply:
the check/5 function ensures that all accounts that could possibly be resolved, are initialized in the accounts tree already, otherwise this would fail